components/markdown/markdown-components.js

```diff
@@ -48,5 +48,24 @@ export default {
   li: ListItem,
   code: ({ node, inline, children, ...props }) =>
     inline ? <code {...props}>{children}</code> : <CodeBlock {...props}>{children}</CodeBlock>,
+
+  // Created a custom pre element which checks if code element is embedded or not.
+  pre: ({ node, children }) => {
+    if (node.children && node.children.length > 0 && node.children[0].tagName === 'code') {
+      return (
+        <pre
+          style={{
+            backgroundColor: 'rgb(40 42 54)',
+            padding: 0,
+            maxWidth: 1024,
+          }}
+        >
+          {children}
+        </pre>
+      );
+    }
+
+    return <pre>{children}</pre>;
+  },
```

Comment:
We can simplify this to something like this.

```suggestion
  pre: ({ node, children }) => <>{children}</>
```

The wrapper doesn't behave very nicely right now on small resolutions, and this should fix the problem.

I don't believe we ever use `pre` without a codeblock.

Reply:
Oh okay, for the case of codeblock itself I used that. This is great. Thanks for the comment

Reply:
alternatively you could do something like

```ts
pre: ({ node, children }) => {
  if (node.children?.length > 0 && node.children[0].tagName === 'code') {
    return children
  }

  return <pre>{children}</pre>
},
```

Reply:
For now i commited the suggestions. I will pull and then add this thing. as

I will do length == 1 as we need only one code block inside one pre right?
``` js
pre: ({ node, children }) => {
  if (node.children?.length ==1 && node.children[0].tagName === 'code') {
    return children
  }

  return <pre>{children}</pre>
},
```

Reply:
I haven't checked the shape of node or children. If you think this is the right way of identifying a codeblock then I'm  good with it too :)

Reply:
Yeah with my previous trials with react-markdown any codeblock is represented as pre>code itself.